### PR TITLE
🏎️🚀 🏎️🚀 🏎️🚀 Test partitioning for MySQL: shave ~5 minutes off of CI runs 🏎️🚀🏎️🚀 🏎️🚀

### LINKED ISSUE #45009: Parallelize slower driver tests
I [tried](https://github.com/metabase/metabase/pull/45392) splitting tests up using simple path filtering using `:only` and `:exclude` directories. This didn't work because of some [limitations](https://github.com/metabase/hawk#excluding-directories) in :exclude-directories.

The right way to handle this is likely to update hawk to support batching tests. (see [thread](https://metaboat.slack.com/archives/CKZEMT1MJ/p1720731827221419))

### PR DESCRIPTION
**Pull request by camsaul**
Closes https://github.com/metabase/metabase/issues/45009

Split off from https://github.com/metabase/metabase/pull/47681

Split up our the slowest CI jobs (MySQL and MariaDB) into two separate jobs (one to run the first half of the tests and one to run the second half of the tests) using the new test partitioning features I added to Hawk in https://github.com/metabase/hawk/pull/28 and https://github.com/metabase/hawk/pull/31 

This turns our ~50 minute jobs into one ~45 minute job and one ~15 minute job. I'm still investigating why the split isn't closer to ~30/~30, I think it's because test run time is dominated by a few certain mega-slow tests (SerDes and migration tests)... once this PR lands I'll experiment with splitting tests up into additional jobs or maybe just splitting the migration tests off into their own jobs

### CODE REVIEW

File: .github/workflows/drivers.yml [hunk 1]
```diff
@@ -522,12 +540,30 @@ jobs:
       uses: ./.github/actions/test-driver
       with:
         junit-name: 'be-tests-mysql-8-0-ee'
+        test-args: >-
+          :partition/total 2
+          :partition/index ${{ matrix.partition-index }}
+    - name: Upload Test Results
```
Comment (camsaul) [inline .github/workflows/drivers.yml:546]: not really part of this PR but this was missing from this job even tho the other jobs all had it

**Comment by piranha** [comment]
Again seems like failures because some tests were not run?

**Review by bshepherdson — APPROVED** [review]
